src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,181 @@
+package wolox.training.controllers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Collections;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+class BookControllerTest {
+
+    private static final String BOOK_PATH = "/api/books";
+    private static final int BOOK_ID = 0;
+    private static final int INVALID_BOOK_ID = 1;
+    private static final int PAGE_NUMBER = 90;
+
+    private MockMvc mockMvc;
+    private BookRepository bookRepository;
+    private ObjectMapper objectMapper;
+    private Book bookTest;
+
+    @BeforeEach
+    void setUp() {
+
+        // Arrange
+        objectMapper = new ObjectMapper();
+        bookRepository = mock(BookRepository.class);
+
+        BookController bookController = new BookController(bookRepository);
+        mockMvc = MockMvcBuilders.standaloneSetup(bookController).build();
```

Comment:
I think you can use the `@WebMvcTest` annotation instead of doing this, but it is ok.